lib/ExpireShare.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace OCA\Testing;
+
+use DateTime;
+use OC\OCS\Result;
+use OCP\IRequest;
+use OCP\Share\Exceptions\GenericShareException;
+use OCP\Share\Exceptions\ShareNotFound;
+use OCP\Share\IManager;
+use OCP\Share\IShare;
+
+class ExpireShare {
+
+	/**
+	 * @var IRequest
+	 */
+	private $request;
+
+
+	/** @var IManager */
+	private $shareManager;
+
+	/**
+	 * @param IRequest $request
+	 * @param IManager $shareManager
+	 */
+	public function __construct(
+		IRequest $request,
+		IManager $shareManager
+	) {
+		$this->request = $request;
+		$this->shareManager = $shareManager;
+	}
+
+	/**
+	 * @param $id
+	 * @return IShare
+	 * @throws ShareNotFound
+	 */
+	private function getShareById($id) {
+		$share = null;
+
+		// First check if it is an internal share.
+		try {
+			$share = $this->shareManager->getShareById('ocinternal:'.$id);
+		} catch (ShareNotFound $e) {
+			if (!$this->shareManager->outgoingServer2ServerSharesAllowed()) {
+				throw new ShareNotFound();
+			}
+
+			$share = $this->shareManager->getShareById('ocFederatedSharing:' . $id);
+		}
+
+		return $share;
+	}
+
+
+	public function expireShare(array $param) {
+		$id = \trim($param["share_id"]);
+		if (!$this->shareManager->shareApiEnabled()) {
+			return new Result(null, 404, 'Share API is disabled');
+		}
+
+		try {
+			$share = $this->getShareById($id);
+			$original_date = $share->getExpirationDate();
+		} catch (ShareNotFound $e) {
+			return new Result(null, 404, 'Wrong share ID, share doesn\'t exist');
+		}
+		// set share expiration to a past datetime
+		$share->setExpirationDate(new \DateTime('yesterday'));
+		$new_date = $share->getExpirationDate();
+		try {
+			$this->shareManager->updateShare($share);
```

Comment:
Note:  this will require a small core change to specially allow tests to `updateShare` with a expiration date in the past.

@kiranparajuli589 can you make a core PR. Then we can discuss with the developers.

Reply:
yes sure @phil-davis

Reply:
PR link https://github.com/owncloud/core/pull/38793